smartconfig/entry.py

```diff
@@ -0,0 +1,123 @@
+from itertools import chain
+from typing import Any, Dict, NoReturn, Optional, Tuple
+
+from smartconfig import _registry
+from smartconfig._registry import global_configuration, get_attribute, used_paths, unload_defaults
+from smartconfig.exceptions import ConfigurationError, ConfigurationKeyError, InvalidOperation, PathConflict
+
+
+class _ConfigEntryMeta(type):
+    """
+    Metaclass used to define special ConfigEntry behaviors.
+
+    Note: Using this metaclass outside of the library is currently not supported.
```

Comment:
I am afraid this will break some internal code that will assume that you are passing a subclass of `ConfigEntry`. I don't mind supporting this, but let's have an MVP first.

Reply:
We may need to rely on subclassing the metaclass for our Discord stuff for all I know, but sure, we can save it for later.